### PR TITLE
Limit identifier characters

### PR DESCRIPTION
**Pull request by Hoverbear**
@eeeebbbbrrrr found a small issue with "ambitiously large" SQL identifier names, eg:

```rust
#[pg_extern]
fn this_function_is_definitely_over_64_characters_because_i_have_given_it_one_as_such_and_this_is_my_desire()  {
    todo!()
}
```

It turns out, PostgreSQL has (by default) a 64 character limit on identifiers, they get truncated after. This is identified by `pgx::pg_sys::NAMEDATALEN`.

This tweaks the SQL generator to staunchly refuse to generate such truncated function names, as it's bad practice.

We weren't able to use `pgx::pg_sys::NAMEDATALEN` since the SQL generator and `pgx::pg_sys` are like oil and water: They can't mix.

### CODE REVIEW

File: pgx-utils/src/sql_entity_graph/postgres_ord/mod.rs [hunk 1]
```diff
@@ -67,17 +67,21 @@ pub struct PostgresOrd {
...
             to_sql_config,
-        }
+        })
     }
 
     pub fn from_derive_input(derive_input: DeriveInput) -> Result<Self, syn::Error> {
         let to_sql_config =
             ToSqlConfig::from_attributes(derive_input.attrs.as_slice())?.unwrap_or_default();
-        Ok(Self::new(derive_input.ident, to_sql_config))
+        Self::new(derive_input.ident, to_sql_config)
```
Comment (workingjubilee) [inline pgx-utils/src/sql_entity_graph/postgres_ord/mod.rs:84]: Wait, how does this work? Shouldn't it still have to Ok-wrap? It's still returning a Result...?

Comment (workingjubilee) [reply pgx-utils/src/sql_entity_graph/postgres_ord/mod.rs:84]: Oh, it's because `::new` is also a Result, okay. I missed that somehow.

Comment (Hoverbear) [reply pgx-utils/src/sql_entity_graph/postgres_ord/mod.rs:84]: Ya! Sorry!

File: pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs [hunk 2]
```diff
@@ -11,46 +11,51 @@ use syn::{ItemFn, Token};
...
+                    let updated_value = value.replace(
+                        "@FUNCTION_NAME@",
+                        &*(func.sig.ident.to_string() + "_wrapper"),
+                    ) + "\n";
+                    *content = syn::LitStr::new(&updated_value, Span::call_site());
+                }
             }
+
+            found.unwrap_or_default()
+        };
```
Comment (workingjubilee) [inline pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs:52]: I found this control flow confusing to follow. Maybe the final unwrap pushed it over the edge for my brain. Would something like this be okay?

```diff
diff --git a/pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs b/pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs
index d4b9e0b..c9143e9 100644
--- a/pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs
+++ b/pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs
@@ -19,37 +19,22 @@ impl PgTrigger {
         func: ItemFn,
         attributes: syn::punctuated::Punctuated<PgTriggerAttribute, Token![,]>,
     ) -> Result<Self, syn::Error> {
-
-        let to_sql_config = {
-            let mut found = None;
-            for attribute in attributes.iter() {
-                match attribute {
-                    &PgTriggerAttribute::Sql(ref to_sql_config) if found.is_none() => {
-                        found = Some(to_sql_config.clone())
-                    }
-                    &PgTriggerAttribute::Sql(_) if found.is_some() => {
-                        return Err(syn::Error::new(
-                            Span::call_site(),
-                            "Multiple `sql` arguments found, it must be unique",
-                        ))
-                    }
-                    _ => (),
-                }
-            }
-
-            if let Some(ref mut found) = found {
-                if let Some(ref mut content) = found.content {
+        if attributes.len() > 1 {
+            return Err(syn::Error::new(
+                Span::call_site(),
+                "Multiple `sql` arguments found, it must be unique",
+            ))
+        };
+        let to_sql_config = attributes.first().cloned().map(|PgTriggerAttribute::Sql(mut config)| {
+                if let Some(ref mut content) = config.content {
                     let value = content.value();
-                    let updated_value = value.replace(
-                        "@FUNCTION_NAME@",
-                        &*(func.sig.ident.to_string() + "_wrapper"),
-                    ) + "\n";
+                    let updated_value = value
+                        .replace("@FUNCTION_NAME@", &*(func.sig.ident.to_string() + "_wrapper"))
+                        + "\n";
                     *content = syn::LitStr::new(&updated_value, Span::call_site());
-                }
-            }
-
-            found.unwrap_or_default()
-        };
+                };
+                config
+            }).unwrap_or_default();
 
         if !to_sql_config.overrides_default() {
             crate::ident_is_acceptable_to_postgres(&func.sig.ident)?;
```

File: pgx-utils/src/sql_entity_graph/to_sql/mod.rs [hunk 3]
```diff
@@ -148,6 +148,19 @@ impl ToSqlConfig {
...
+        if self.enabled == false {
+            return true
+        }
+        if self.callback.is_some() {
+            return true
+        }
+        if self.content.is_some() {
+            return true
+        }
+        false
```
Comment (workingjubilee) [inline pgx-utils/src/sql_entity_graph/to_sql/mod.rs:162]:
```suggestion
        self.enabled == false || self.callback.is_some() || self.content.is_some()
```

**Review by workingjubilee — APPROVED** [review]
Modulo that, everything looks good!

File: pgx-utils/src/sql_entity_graph/to_sql/mod.rs (hunk 3 above)
Comment (Hoverbear) [reply pgx-utils/src/sql_entity_graph/to_sql/mod.rs:162]: :golf: :grinning:

**Pull request merged**